src/helpers/turbo.ts

```diff
@@ -0,0 +1,100 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import snapshot from '@snapshot-labs/snapshot.js';
+import db from './mysql';
+import { fetchWithKeepAlive } from './utils';
+
+type Space = {
+  id: string;
+  turbo_expiration: number;
+};
+
+const SCHNAPS_API_URL = process.env.SCHNAPS_API_URL;
+const NETWORK_PREFIX = process.env.NETWORK === '1' ? 's:' : 's-tn:';
+const RUN_INTERVAL = 10 * 1e3; // 10 seconds
+
+// Periodically sync the turbo status of spaces with the schnaps-api
+export async function trackTurboStatuses() {
+  if (!SCHNAPS_API_URL) return;
+
+  while (true) {
+    // Step 1: Query all the spaces from the schnaps-api
+    let spaces = await getSpacesExpirationDates();
+
+    spaces = spaces
+      .filter(space => space.id.startsWith(NETWORK_PREFIX))
+      .map(space => {
+        return { ...space, id: space.id.replace(NETWORK_PREFIX, '') };
+      });
+
+    // Step 2: Update the turbo status of the spaces in the database
+    updateTurboStatuses(spaces);
+
+    // Sleep for a while before running the loop again
+    await snapshot.utils.sleep(RUN_INTERVAL);
+  }
+}
+
+async function updateTurboStatuses(spaces: { id: string; turbo_expiration: number }[]) {
+  if (spaces.length === 0) return;
+
+  const now = Math.floor(Date.now() / 1000); // Current timestamp in seconds
+
+  // Sync `turbo_expiration` and `turbo` status for each space
+  const query = `
+    UPDATE spaces
+    SET
+      turbo_expiration = CASE id
+        ${spaces.map(() => `WHEN ? THEN ?`).join(' ')}
+      END,
+      turbo = CASE
+        WHEN turbo_expiration < ? THEN 0
+        ELSE 1
+      END
+    WHERE id IN (${spaces.map(() => '?').join(', ')});
+  `;
+
+  // Flatten `spaces` array: [id1, expiration1, id2, expiration2, ..., now, id1, id2, ...]
+  const params = spaces.flatMap(({ id, turbo_expiration }) => [id, turbo_expiration]);
+
+  // Add `now` for the `CASE` condition
+  params.push(now);
+
+  // Append `id`s again for the `WHERE IN (...)` clause
+  params.push(...spaces.map(space => space.id));
+
+  await db.query(query, params);
+}
+
+async function getSpacesExpirationDates(): Promise<Space[]> {
+  const query = `
+    query GetSpaces {
+      spaces {
+        id
+        turbo_expiration
+      }
+    }
+  `;
+
+  try {
+    const response = await fetchWithKeepAlive(SCHNAPS_API_URL, {
+      timeout: 5e3,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ query })
+    });
+
+    const data = await response.json();
```

Comment:
Not too important (for now), but if there are thousands of turbo spaces, will it return all of them without any limits?

Reply:
Yes, once we have thousand of Turbo spaces (which should give us a bit of time) we can change that logic.

Reply:
@pscott what's the pagination default and max limit?